### PR TITLE
Return BlockForRpc from debug_getBadBlocks

### PR DESCRIPTION
**Pull request by flcl42**
## Changes

- Fix return type for debug_getBadBlocks

#### What types of changes does your code introduce?

- [x] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [ ] Yes
- [x] No

### Notes on testing

1. Enable debug module
2. `curl --request POST --url http://127.0.0.1:8545 --header 'content-type: application/json' --data '{ "method": "debug_getBadBlocks", "params": [], "id": 1, "jsonrpc": "2.0" }'`

### CODE REVIEW

File: src/Nethermind/Nethermind.JsonRpc/Modules/Eth/DebugBlockForRpc.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+using Nethermind.Core.Specs;
+using Nethermind.Core;
+using Nethermind.Serialization.Rlp;
+
+namespace Nethermind.JsonRpc.Modules.Eth;
+
+public class DebugBlockForRpc(Block block, bool includeFullTransactionData, ISpecProvider specProvider, HeaderDecoder headerDecoder) : BlockForRpc(block, includeFullTransactionData, specProvider)
+{
+    public byte[] Rlp { get; } = headerDecoder.Encode(block.Header).Bytes;
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.JsonRpc/Modules/Eth/DebugBlockForRpc.cs:12]: WHy only header RLP? Is there other way to get bad block RLP?

Comment (flcl42) [reply src/Nethermind/Nethermind.JsonRpc/Modules/Eth/DebugBlockForRpc.cs:12]: Docs mention header, code of others tells about entire block. Let me figure out

Comment (flcl42) [reply src/Nethermind/Nethermind.JsonRpc/Modules/Eth/DebugBlockForRpc.cs]: Fixed

**Comment by LukaszRozmej** [comment]
Spec: 
https://github.com/ethereum/execution-apis/blob/cea7eeb642052f4c2e03449dc48296def4aafc24/src/debug/getters.yaml#L82
https://github.com/ethereum/execution-apis/blob/cea7eeb642052f4c2e03449dc48296def4aafc24/src/schemas/block.yaml#L140-L157

**Review by LukaszRozmej — APPROVED** [review]

**Pull request merged**